server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamFailureStoreDefinition.java

```diff
@@ -19,19 +20,33 @@
 import org.elasticsearch.index.mapper.RoutingFieldMapper;
 
 import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
 
 /**
  * A utility class that contains the mappings and settings logic for failure store indices that are a part of data streams.
  */
 public class DataStreamFailureStoreDefinition {
 
     public static final String FAILURE_STORE_REFRESH_INTERVAL_SETTING_NAME = "data_streams.failure_store.refresh_interval";
+    public static final String INDEX_FAILURE_STORE_VERSION_SETTING_NAME = "index.failure_store.version";
     public static final Settings DATA_STREAM_FAILURE_STORE_SETTINGS;
+    // Only a subset of user configurable settings is applicable for a failure index. Here we have an
+    // allowlist that will filter all other settings out.
+    public static final Set<String> SUPPORTED_USER_SETTINGS = Set.of(
+        DataTier.TIER_PREFERENCE,
+        IndexMetadata.SETTING_INDEX_HIDDEN,
+        INDEX_FAILURE_STORE_VERSION_SETTING_NAME,
+        IndexMetadata.SETTING_NUMBER_OF_SHARDS,
+        IndexMetadata.SETTING_NUMBER_OF_REPLICAS,
+        IndexMetadata.SETTING_AUTO_EXPAND_REPLICAS,
+        IndexSettings.INDEX_REFRESH_INTERVAL_SETTING.getKey()
+    );
```

Comment:
Of course you are right, ILM policy should be added.

About the `_tier_preference` I have been going back and forth. Do we want to copy this from the template? Or should we start without it and then ILM can set the settings later? If you think it's safer to copy them, then I will. I do not have strong feelings about it. I am just not sure

Reply:
Nvm, now I get what you mean, if we include `_tier_preference ` we should include the other settings too because they configure a similar behaviour. You are right. Ignore my previous comment then. I will fix it.